website/app/components/doc/page/header/search/index.js

```diff
@@ -0,0 +1,207 @@
+        if (!state.isOpen && prevState.isOpen) {
+          autocompleteInstance.setQuery('');
+        }
+      },
+
+      //
+      // GET SOURCES
+      // see: https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-js/autocomplete/#param-getsources
+      getSources({ query }) {
+        if (query.trim() !== '') {
+          return [
+            {
+              sourceId: 'content',
+              getItems: getItemsFunction({
+                searchQuery: query,
+                searchParams: {
+                  hitsPerPage: 5,
+                  attributesToHighlight: ['pageTitle', 'content'],
+                  attributesToSnippet: ['content:20'],
+                  distinct: true,
+                  // attributeForDistinct: 'pageBaseURL', // this doesn't work, for some reason; it needs to be set at Algolia dashboard level under "Deduplication and Grouping" configuration
+                },
+                searchFilters: { facetFilters: ['type:-icon', 'type:-token'] },
+              }),
+              getItemUrl: ({ item }) => {
+                return item.searchResultURL;
+              },
+              templates: {
+                header: templatesHeaderFunction({ group: 'generic' }),
+                // footer: templatesFooterFunction,
+                item: templatesItemFunction(),
+              },
+            },
+            {
+              sourceId: 'icons',
+              getItems: getItemsFunction({
+                searchQuery: query,
+                searchParams: {
+                  hitsPerPage: 5,
+                  attributesToHighlight: ['icon-name'],
+                  typoTolerance: false,
+                },
+                searchFilters: { facetFilters: ['type:icon'] },
+              }),
+              getItemUrl: ({ item }) => {
+                return item.searchResultURL;
+              },
+              templates: {
+                header: templatesHeaderFunction({ group: 'icons' }),
+                // footer: templatesFooterFunction,
+                item: templatesItemFunction(),
+              },
+            },
+            {
+              sourceId: 'tokens',
+              getItems: getItemsFunction({
+                searchQuery: query,
+                searchParams: {
+                  hitsPerPage: 5,
+                  attributesToHighlight: ['token-name'],
+                  typoTolerance: false,
+                },
+                searchFilters: { facetFilters: ['type:token'] },
+              }),
+              getItemUrl: ({ item }) => {
+                return item.searchResultURL;
+              },
+              templates: {
+                header: templatesHeaderFunction({ group: 'tokens' }),
+                // footer: templatesFooterFunction,
+                item: templatesItemFunction(),
+              },
+            },
+          ];
+        } else {
+          return [
```

Comment:
this is shown when the query is empty. we can customize what we want to show, in case